pkg/services/ngalert/notifier/channels/slack_test.go

```diff
@@ -271,6 +412,12 @@ func TestSlackNotifier(t *testing.T) {
 			require.NoError(t, err)
 
 			require.JSONEq(t, string(expBody), body)
+
+			if c.webhookSent {
+				require.True(t, len(notificationService.Webhook.Url) > 0)
```

Comment:
In this case we're asserting that we tried to upload the image file, correct? Should we compare the URL against the actual expected URL for this endpoint? This is less brittle than it seems if we use `Contains` and compare against a constant, like `require.Contains(t, notificationService.Webhook.Url, slackUploadImageUrl)` maybe?

Reply:
I like it

Reply:
I ended up injecting the expected URL so we could test the custom image domain case too.